File: Problem 39/ans.py
L=[]
for p in range(0,1001):
    print(p)
    cnt=0
    for a in range(0,p+1):
        for b in range(0,a+1):
            for c in range(0,a+b+1):
                if (a*a==(b*b+c*c)) and a+b+c==p:
                    cnt+=1

    L.append(cnt)
print("Solutions maximised when p=%d"%L.index(max(L)))

File: Problem 39/ans.cpp
#include<iostream>

using namespace std;

int main()
{
    long int tot1,tot2;
    int a,b,c,p,cnt,hi=0,pmax;
    for(p=0; p<=1000; p++)
    {
        cnt=0;
        for(c=0; c<=p; c++)
            for(b=0; b<=c; b++)
                for(a=0; a<=b; a++)
                {
                    tot1=c*c;
                    tot2=a*a+b*b;
                    if (tot1==tot2 && a+b+c==p)
                        cnt++;
                }
        if (cnt>hi)
        {
            hi=cnt;
            pmax=p;
        }
    }
    cout << "Number of solutions maximised when p=" << pmax << endl;
    return 0;
}
